src/EnergyPlus/WindTurbine.cc

```diff
@@ -263,12 +275,11 @@ namespace WindTurbine {
                 }
             }
             // Select rotor type
-            {
-                auto const SELECT_CASE_var(state.dataIPShortCut->cAlphaArgs(3));
-                if ((SELECT_CASE_var == "HORIZONTALAXISWINDTURBINE") || (SELECT_CASE_var == "")) {
+            state.dataWindTurbine->WindTurbineSys(WindTurbineNum).rotorType =
+                static_cast<RotorType>(getEnumerationValue(WindTurbine::RotorNamesUC, state.dataIPShortCut->cAlphaArgs(3)));
+            if (state.dataWindTurbine->WindTurbineSys(WindTurbineNum).rotorType == RotorType::Invalid) {
+                if (state.dataIPShortCut->cAlphaArgs(3).empty()) {
```

Comment:
This case gets checked only if the return value of `getEnumerationValue()` is `RotorType::Invalid`. Which happens if the user input string does not match with any of the elements in `RotorNamesUC`.